### PR TITLE
278 add search criteria lookups

### LINKED ISSUE #278: Add Search criteria to filter folder in pipapi::create_versioned_lkups
For testing purposes, we need to be able to look up in one or some of the folder available in our internal drives. The function `pipapi::create_versioned_lkups` currently takes too long to search throughout all the available folders, which makes it inconvenient for efficiency. A simple regular expression call could filter de desired folder. 

## Prework

* [x] If there is [already a relevant issue](https://github.com/PIP-Technical-Team/pipapi/issues), whether open or closed, comment on the existing thread instead of posting a new issue.
* [x] New features take time and effort to create, and they take even more effort to maintain. So if the purpose of the feature is to resolve a struggle you are encountering personally, please consider first posting a "question" or "other" issue so we can discuss your use case and search for existing solutions first.
* [x] Format your code according to the [tidyverse style guide](https://style.tidyverse.org/).

## Proposal

If applicable, write a minimal example in R code or pseudo-code to show input, usage, and desired output.

To help us read any code you include (optional) please try to follow the [tidyverse style guide](https://style.tidyverse.org/). The `style_text()` and `style_file()` functions from the [`styler`](https://github.com/r-lib/styler) package make it easier.

### PR DESCRIPTION
**Pull request by randrescastaneda**
# Prework

* [x] I understand and agree to the [contributing guidelines](https://github.com/PIP-Technical-Team/pipapi/blob/master/CONTRIBUTING.md).
* [x] I have already submitted an [issue](https://github.com/PIP-Technical-Team/pipapi/issues) to discuss my idea with the maintainer.

# Related GitHub issues and pull requests

* close #278 

# Summary
Hi Tony (@tonyfujs), 

I added a small function to create a proper vintage call to `get_vintage_pattern_regex()`. The current version of `create_versioned_lkups()` calls `vintage_pattern = get_vintage_pattern_regex()` by default, but given that `get_vintage_pattern_regex()` return a list, it is very difficult to do the proper specification of the regular expression. The new function `create_vintage_pattern_call()` receives `NULL`, a `character` vector, or a `list`, and parses it properly to `get_vintage_pattern_regex()`.  Now, the default version of `create_versioned_lkups` is like you see below, *but its usability remains the same.* 

```r
create_versioned_lkups <- function(data_dir, vintage_pattern = NULL) 
```

We can keep using `create_versioned_lkups()` as before, but now the `vintage_pattern`  parameter allows to make any regex call to get only the version we want to work with. 

## Examples of new function `create_vintage_pattern_call`. 
``` r
library(pipapi)
vintage_pattern <- NULL
pipapi:::create_vintage_pattern_call(vintage_pattern)
#> $vintage_pattern
#> [1] "\\d{8}_\\d{4}_\\d{2}_\\d{2}_(PROD|TEST|INT)$"
#> 
#> $prod_regex
#> [1] "PROD$"
#> 
#> $int_regex
#> [1] "INT$"
#> 
#> $test_regex
#> [1] "TEST$"

vintage_pattern <- list("r.*", "", "^hjkhj\\.d")
pipapi:::create_vintage_pattern_call(vintage_pattern)
#> $vintage_pattern
#> [1] "r.*"
#> 
#> $prod_regex
#> [1] ""
#> 
#> $int_regex
#> [1] "^hjkhj\\.d"
#> 
#> $test_regex
#> [1] "TEST$"

vintage_pattern <- c("r.*", "", "^hjkhj\\.d")
pipapi:::create_vintage_pattern_call(vintage_pattern)
#> $vintage_pattern
#> [1] "r.*"
#> 
#> $prod_regex
#> [1] ""
#> 
#> $int_regex
#> [1] "^hjkhj\\.d"
#> 
#> $test_regex
#> [1] "TEST$"

vintage_pattern <- c(vintage_pattern = "r.*", test_regex = "", int_regex =  "^hjkhj\\.d")
pipapi:::create_vintage_pattern_call(vintage_pattern)
#> $vintage_pattern
#> [1] "r.*"
#> 
#> $prod_regex
#> [1] "PROD$"
#> 
#> $int_regex
#> [1] "^hjkhj\\.d"
#> 
#> $test_regex
#> [1] ""
```

<sup>Created on 2022-08-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### CODE REVIEW

File: R/create_lkups.R [hunk 1]
```diff
@@ -282,15 +284,115 @@ create_lkups <- function(data_dir, versions) {
...
+
+      get_vintage_pattern_regex()
+
+    } else {
+
+
+      lf <-
+        formals(get_vintage_pattern_regex) |>
+        names() |>
+        length()
```
Comment (tonyfujs) [inline R/create_lkups.R:368]: The R version we use in PROD does not yet support the native pipe I believe. We could update it, but for the time being, let's not use the native pipe

File: R/create_lkups.R [hunk 2]
```diff
@@ -282,15 +284,115 @@ create_lkups <- function(data_dir, versions) {
...
+#' create_vintage_pattern_call(vintage_pattern)
+#'
+#' vintage_pattern <- c(vintage_pattern = "r.*", test_regex = "", int_regex =  "^hjkhj\\.d")
+#' create_vintage_pattern_call(vintage_pattern)
+#' }
+create_vintage_pattern_call <- function(vintage_pattern = NULL) {
+
+  #   ______________________________________________________________________
+  #   on.exit                                               ####
+  on.exit({
```
Comment (tonyfujs) [inline R/create_lkups.R:338]: I'm not sure I follow everything that is happening here. Maybe we can go over it when we meet later today?

**Review by tonyfujs — CHANGES_REQUESTED** [review]
I really like this idea. Let's discuss my questions when we meet today.

File: R/create_lkups.R [hunk 3]
```diff
@@ -282,15 +284,115 @@ create_lkups <- function(data_dir, versions) {
...
+                             prod_regex),
+
+    int_regex       = ifelse(is.null(int_regex),
+                             "INT$",
+                             int_regex),
+
+    test_regex      = ifelse(is.null(test_regex),
+                             "TEST$",
+                             test_regex)
     )
```
Comment (tonyfujs) [inline R/create_lkups.R:310]: I think vectorized `ifelse` is slower than regular `if... else`. May be worth updating if we are always expecting arguments of length 1.

Comment (randrescastaneda) [reply R/create_lkups.R]: Done. This branch is also working on the dev server.

**Pull request merged**